harvester/src/test/java/com/github/reflectoring/infiniboard/harvester/scheduling/SchedulingServiceTest.java

```diff
@@ -0,0 +1,62 @@
+package com.github.reflectoring.infiniboard.harvester.scheduling;
+
+import com.github.reflectoring.infiniboard.packrat.source.SourceConfig;
+import org.apache.commons.lang3.mutable.MutableInt;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.quartz.*;
+
+import java.util.HashMap;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * tests the scheduling service by using TestJob
+ */
+public class SchedulingServiceTest {
+
+    public static final String TEST_JOB = "TestJob";
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Test
+    public void notRegisteringTwice() throws SchedulerException {
+        SchedulingService schedulingService = new SchedulingService();
+        schedulingService.registerJob(TEST_JOB, TestJob.class);
+        expectedException.expectMessage("job type TestJob is already registered by " + TestJob.class.toString());
+        schedulingService.registerJob(TEST_JOB, TestJob.class);
+    }
+
+    @Test
+    public void scheduleJob() throws SchedulerException, InterruptedException {
```

Comment:
let's name our test methods with a pattern. anyone should be able to know what the test is expected to check by reading the test method name.

I prefer some pattern like this: scheduleJobShouldRunAtLeastThreeTimes